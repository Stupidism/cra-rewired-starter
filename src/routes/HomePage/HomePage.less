.HomePage {
  font-size: large;
  padding: 20px;
}
