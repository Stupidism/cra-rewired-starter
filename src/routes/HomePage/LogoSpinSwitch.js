import React from 'react';
import { Switch } from 'antd';
import { compose, withHandlers } from 'recompose';

import logoSpinning from 'store/selectors/logoSpinning';
import connect from 'store/connect';
import { setRuntime } from 'store/modules/runtime';

export const LogoSpinSwitch = ({ logoSpinning, toggleLogoSpin }) => (
  <Switch checked={logoSpinning} onChange={toggleLogoSpin} />
);

export const toggleLogoSpin = ({ logoSpinning, dispatch }) => () =>
  dispatch(setRuntime({ logoSpinning: !logoSpinning }));

export default compose(
  connect({ logoSpinning }),
  withHandlers({ toggleLogoSpin }),
)(LogoSpinSwitch);
