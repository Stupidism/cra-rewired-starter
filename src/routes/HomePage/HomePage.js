import React from 'react';
import { Link } from 'react-router-dom';

import ExternalLink from 'components/ExternalLink';

import LogoSpinSwitch from './LogoSpinSwitch';
import './HomePage.less';

export const HomePage = () => (
  <div className="HomePage">
    <p>
      To get started, edit <code>src/containers/App/App.js</code> and save to
      reload.
    </p>
    <p>
      This repo is created upon{' '}
      <ExternalLink to="https://github.com/facebook/create-react-app">
        create-react-app
      </ExternalLink>. Extra features are listed below:
    </p>
    <p>
      Click <Link to="/some/undefined/route">some undefined route</Link> to try{' '}
      <ExternalLink to="https://github.com/ReactTraining/react-router">
        react-router
      </ExternalLink>.
    </p>
    <p>
      Look at this beautiful toggle <LogoSpinSwitch />, it's from{' '}
      <ExternalLink to="https://ant.design/components/switch/">
        ant-design
      </ExternalLink>.
    </p>
    <p>
      Did you find? The toggle controls the logo across components thanks to{' '}
      <ExternalLink to="https://redux.js.org/introduction">redux</ExternalLink>.
    </p>
  </div>
);

export default HomePage;
