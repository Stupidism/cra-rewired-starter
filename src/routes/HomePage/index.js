import createAsyncComponent from 'modules/createAsyncComponent';

export default createAsyncComponent(() =>
  import(/* webpackChunkName: "routes.HomePage" */ './HomePage'),
);
