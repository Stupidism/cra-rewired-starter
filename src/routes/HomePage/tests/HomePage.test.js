import React from 'react';
import { shallow } from 'enzyme';

import HomePage from '../HomePage';

describe('<HomePage />', () => {
  it('should be consistent with snapshot', () => {
    const snapshot = shallow(<HomePage />);
    expect(snapshot).toMatchSnapshot();
  });
});
