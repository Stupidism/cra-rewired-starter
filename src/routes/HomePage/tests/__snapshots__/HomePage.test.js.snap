// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should be consistent with snapshot 1`] = `
<div
  className="HomePage"
>
  <p>
    To get started, edit 
    <code>
      src/containers/App/App.js
    </code>
     and save to reload.
  </p>
  <p>
    This repo is created upon
     
    <Component
      to="https://github.com/facebook/create-react-app"
    >
      create-react-app
    </Component>
    . Extra features are listed below:
  </p>
  <p>
    Click 
    <Link
      replace={false}
      to="/some/undefined/route"
    >
      some undefined route
    </Link>
     to try
     
    <Component
      to="https://github.com/ReactTraining/react-router"
    >
      react-router
    </Component>
    .
  </p>
  <p>
    Look at this beautiful toggle 
    <Connect(withHandlers(Component)) />
    , it's from
     
    <Component
      to="https://ant.design/components/switch/"
    >
      ant-design
    </Component>
    .
  </p>
  <p>
    Did you find? The toggle controls the logo across components thanks to
     
    <Component
      to="https://redux.js.org/introduction"
    >
      redux
    </Component>
    .
  </p>
</div>
`;
